client/handle_http.go

```diff
@@ -2153,6 +2153,14 @@ func uploadObject(transfer *transferFile) (transferResult TransferResults, err e
 		ioreader = ap
 		sizer = ap
 	} else {
+
+		if fileInfo.IsDir() {
+			log.Errorln("The given path", transfer.localPath, "is a directory. Use --recursive or -r flag if using the pelican object command for directories")
+			err := errors.New("the provided path '" + transfer.localPath + "' is a directory, but a file is expected; use --recursive or -r flag if using the pelican object command for directories")
```

Comment:
One quick comment since I'm looking at this now -- instead of having two extremely similar messages, one that gets logged and one that gets returned, just return the error. Otherwise the debug logs at the client get very repetitive.

Reply:
Yes, I understand. I have updated this in my latest commit and am now only returning the error.

https://github.com/PelicanPlatform/pelican/pull/1707/files#diff-4b3ff5f5414bcb82345ca02dead65d984f23d0737392da402d2517b8f192ce61R2158-R2160